pylint/config/arguments_manager.py

```diff
@@ -145,15 +146,17 @@ def _parse_configuration_file(self, config_data: Dict[str, str]) -> None:
         # TODO: This should parse_args instead of parse_known_args
         self.namespace = self._arg_parser.parse_known_args(arguments, self.namespace)[0]
 
-    def _parse_command_line_configuration(self, arguments: List[str]) -> None:
+    def _parse_command_line_configuration(
+        self, arguments: Optional[List[str]] = None
+    ) -> List[str]:
         """Parse the arguments found on the command line into the namespace."""
-        # pylint: disable-next=fixme
-        # TODO: This should parse_args instead of parse_known_args
-        self.namespace = self._arg_parser.parse_known_args(arguments, self.namespace)[0]
+        arguments = sys.argv[1:] if arguments is None else arguments
```

Comment:
I think we should force this function to have arguments (``def _parse_command_line_configuration(self, arguments: List[str]) ->List[str]``). Imo the ``sys.argv[1:] if arguments is None else arguments`` should be done in only one place: for the default value of the API entrypoint and then we have list of strings with no default value allowed everywhere else. This way we make sure that we have a clearly defined API for our internal functions, no special case to handle for None value and we have only one place where we define the arguments.

Reply:
https://github.com/PyCQA/pylint/blob/48e658551308daddf52fb62483356e39a3de097e/pylint/config/option_manager_mixin.py#L317-L332

Also does this, so in order not to break compatibility with it I thought I would copy the approach there...
I'm not sure what's best here. I have seen that in `pyreverse` we actually do this as well but before we call `load_command_line_configuration`. For `pylint` I think there are 3 places where we do this 😅 

Perhaps we could add a deprecation warning after everything else has been done?

Reply:
>  we could add a deprecation warning after everything else has been done?

Seems sensible.